params/astria_config.go

```diff
@@ -0,0 +1,369 @@
+					return nil, errors.New("astria bridge sender address must be set for bridged ERC20 assets")
+				}
+
+				bridgeCfg := cfg
+				orderedForks[i].BridgeAddresses[cfg.BridgeAddress] = &bridgeCfg
+				orderedForks[i].BridgeAllowedAssets[cfg.AssetDenom] = struct{}{}
+				if cfg.Erc20Asset == nil {
+					log.Info("bridge for sequencer native asset initialized", "bridgeAddress", cfg.BridgeAddress, "assetDenom", cfg.AssetDenom)
+				} else {
+					log.Info("bridge for ERC20 asset initialized", "bridgeAddress", cfg.BridgeAddress, "assetDenom", cfg.AssetDenom, "contractAddress", cfg.Erc20Asset.ContractAddress)
+				}
+			}
+		}
+	}
+
+	if err := validateAstriaForks(orderedForks); err != nil {
+		return nil, err
+	}
+
+	// Sort orderedForks in descending order by Height
+	sort.Slice(orderedForks, func(i, j int) bool {
+		return orderedForks[i].Height > orderedForks[j].Height
+	})
+
+	return &AstriaForks{
+		orderedForks: orderedForks,
+		forkMap:      forks,
+	}, nil
+}
+
+func validateAstriaForks(forks []AstriaForkData) error {
+	for _, fork := range forks {
+		if !fork.Halt {
+			if fork.Sequencer.ChainID == "" {
+				return fmt.Errorf("fork %s: sequencer chain ID not set", fork.Name)
+			}
+
+			if fork.Sequencer.StartHeight == 0 {
+				return fmt.Errorf("fork %s: sequencer initial height not set", fork.Name)
+			}
+
+			if fork.Celestia.ChainID == "" {
+				return fmt.Errorf("fork %s: celestia chain ID not set", fork.Name)
+			}
+
+			if fork.Celestia.StartHeight == 0 {
+				return fmt.Errorf("fork %s: celestia initial height not set", fork.Name)
+			}
+
+			if fork.Celestia.HeightVariance == 0 {
+				return fmt.Errorf("fork %s: celestia height variance not set", fork.Name)
+			}
+
+			if fork.FeeCollector == (common.Address{}) {
+				log.Warn("fee asset collectors not set, assets will be burned", "fork", fork.Name)
+			}
+		} else {
+			log.Warn("fork will halt", "fork", fork.Name, "height", fork.Height)
+		}
+	}
+
+	return nil
+}
+
+func GetDefaultAstriaForkData() AstriaForkData {
+	return AstriaForkData{
+		Height:       1,
+		FeeCollector: common.Address{},
+		EIP1559Params: AstriaEIP1559Params{
+			MinBaseFee:               0,
+			ElasticityMultiplier:     DefaultElasticityMultiplier,
+			BaseFeeChangeDenominator: DefaultBaseFeeChangeDenominator,
+		},
+		BridgeAddresses:     make(map[string]*AstriaBridgeAddressConfig),
+		BridgeAllowedAssets: make(map[string]struct{}),
+	}
+}
+
+func (c *AstriaForks) GetForkAtHeight(height uint64) AstriaForkData {
+	idx := sort.Search(len(c.orderedForks), func(i int) bool {
+		return c.orderedForks[i].Height <= height
+	})
+	// no named fork at this height
+	if idx == len(c.orderedForks) {
+		return GetDefaultAstriaForkData()
+	}
+	return c.orderedForks[idx]
+}
+
+func (c *AstriaForks) GetNextForkAtHeight(height uint64) *AstriaForkData {
+	idx := sort.Search(len(c.orderedForks), func(i int) bool {
+		return c.orderedForks[i].Height > height
+	})
```

Comment:
if there are multiple forks higher than `height`, this will return the overall highest, not the one directly higher than `height` (since `orderedForks` is descending by height)

Reply:
yep. will fix